File: Problem/Easy/Problem_9/Solution.py
# Python3 program using to find sum of its digits using recursion.

# Recursive function
def sumDigits(num : int, total = 0) -> int:
    if num == 0:
        return total
    return sumDigits(int(num // 10), total + num % 10)
    
# Main
if __name__ == '__main__':
    n = 45632
    print("Sum digits of {} is: {}".format(n, sumDigits(n)))

File: Problem/Easy/Problem_9/Solution.cpp
// C++ program using to find sum of its digits using recursion.

#include <iostream>

// Recursive function
int sum_digits(int n, int sum = 0)
{
    if (n == 0)
        return sum;
    return sum_digits(n/10, sum+n%10);
}

// Main function
int main()
{
    int n = 45632;
    std::cout << "Sum digits of " << n << " is: " << sum_digits(n);
    return 0;
}